core/controllers/base_test.py

```diff
@@ -288,6 +293,25 @@ def test_keys_match_en_qqq(self):
         qqq_key_list = self._extract_keys_from_json_file('qqq.json')
         self.assertEqual(en_key_list, qqq_key_list)
 
+    def test_keys_in_source_code_match_en(self):
+        """Tests that keys in HTML files are present in en.json."""
+        en_key_list = self._extract_keys_from_json_file('en.json')
+        for root, _, files in os.walk(os.path.join(
+                os.getcwd(), self.get_static_asset_filepath())):
+            for filename in files:
+                if filename.endswith('.html'):
+                    html_key_list = self._extract_keys_from_html_file(
+                        os.path.join(root, filename))
+                    self.assertLessEqual(set(html_key_list), set(en_key_list))
+                    if set(html_key_list) > set(en_key_list):
```

Comment:
I don't think you actually reach this part of the code, due to the assertion failure? Perhaps, instead, you can add a custom error message to the assertion -- something like "Missing keys in en.json: I18N_blah_blah_blah found in file blah.html". You can test the error case by putting a bad key in some html file and seeing if the error message that's printed out is easily comprehensible.